ethstats/ethstats.go

```diff
@@ -382,8 +415,12 @@ func (s *Service) login(conn *websocket.Conn) error {
 		network = fmt.Sprintf("%d", infos.Protocols["les"].(*les.NodeInfo).Network)
 		protocol = fmt.Sprintf("les/%d", les.ClientProtocolVersions[0])
 	}
+	if s.eth != nil {
+		etherBase, _ = s.eth.Etherbase()
+	}
 	auth := &authMsg{
-		ID: s.node,
+		ID:      s.node,
+		Address: etherBase,
```

Comment:
Does it matter that this could be nil if `s.eth` is nil up above? If so should we fail?